frontend/src/pages/modelCatalog/screens/ModelDetailsPage.tsx

```diff
@@ -42,18 +42,20 @@ const ModelDetailsPage: React.FC = conditionalArea(
           objectType={ProjectObjectType.registeredModels}
         />
       }
-      empty={model === null}
+      empty={Boolean(modelCatalogSources.error) || model === null}
       emptyStatePage={
         <EmptyModelCatalogState
           testid="empty-model-catalog-state"
-          title="Details not found"
-          description="To request access to model catalog, contact your administrator."
+          title={modelCatalogSources.error ? 'Details not found' : 'Unable to load model details'}
+          description={modelCatalogSources.error?.message || 'Refresh the page or try again later'}
           headerIcon={() => (
-            <img src={typedEmptyImage(ProjectObjectType.registeredModels)} alt="" />
+            <img src={typedEmptyImage(ProjectObjectType.registeredModels, 'Error')} alt="" />
           )}
         />
```

Comment:
Similarly here, we don't need to do any of this error rendering in emptyStatePage. You can put this all back as it was - it is only used if there are no model catalog sources.

Aha -- I know why your mysterious non-404 cypress test was passing when it shouldn't. You're rendering `'Unable to load model details'` here in the empty state even though there is no error. If you revert this stuff, that test will fail as it should. I'll edit the earlier comment.